dvc/command/experiments.py

```diff
@@ -2,14 +2,57 @@
 import io
 import logging
 from collections import OrderedDict
+from itertools import groupby
+from typing import Iterable, Optional
 
 from dvc.command.base import CmdBase, append_doc_link, fix_subparsers
 from dvc.command.metrics import DEFAULT_PRECISION
-from dvc.exceptions import DvcException
+from dvc.exceptions import DvcException, InvalidArgumentError
 
 logger = logging.getLogger(__name__)
 
 
+def _filter_names(
+    names: Iterable,
+    label: str,
+    include: Optional[Iterable],
+    exclude: Optional[Iterable],
+):
+    if include and exclude:
+        intersection = set(include) & set(exclude)
+        if intersection:
+            values = ", ".join(intersection)
+            raise InvalidArgumentError(
+                f"'{values}' specified in both --include-{label} and"
+                f" --exclude-{label}"
+            )
+
+    names = {tuple(name.split(".")) for name in names}
+
+    def _filter(filters, update_func):
+        filters = {tuple(name.split(".")) for name in filters}
+        for length, groups in groupby(filters, len):
+            for group in groups:
+                matches = [name for name in names if name[:length] == group]
```

Comment:
If we are going to the extent of matching on high level, maybe we could consider support for expressions?
Just an idea, and not for this change, I am pretty sure that parsing from the command line will be a headache if we were to go this way.

Reply:
Yeah I was thinking about this too. Support for more complex expressions seems like a nice to have we could consider adding down the line, but would be overkill for now.